astropy/coordinates/builtin_frames/utils.py

```diff
@@ -236,25 +279,61 @@ def atciqz(rc, dc, astrom):
     same approach used by the ERFA functions above, except that they use a threshold of
     9 arcseconds.
 
+    In addition, ERFA's atciqz assumes a distant source, so there is no difference between
+    the object-Sun vector and the observer-Sun vector. This can lead to errors of up to a
+    few arcseconds in the worst case (e.g a Venus transit).
+
     Parameters
     ----------
-    rc : float or `~numpy.ndarray`
-        right ascension, radians
-    dc : float or `~numpy.ndarray`
-        declination, radians
+    astrometric : `~astropy.coordinates.CartesianRepresentation`
+        Astrometric ICRS position of object from observer
     astrom : eraASTROM array
         ERFA astrometry context, as produced by, e.g. ``eraApci13`` or ``eraApcs13``
 
     Returns
     --------
     ri : float or `~numpy.ndarray`
+        Right Ascension in radians
     di : float or `~numpy.ndarray`
+        Declination in radians
     """
+    # first of all get distance from object to observer
+    object_observer_distance = astrometric.norm()
+    # ignore parallax effects if no distance, or far away
+    ignore_distance = object_observer_distance.unit == u.one
+
     # BCRS coordinate direction (unit vector).
-    pco = erfa.s2c(rc, dc)
+    if not ignore_distance and u.allclose(astrometric.xyz, 0*u.km):
```

Comment:
This will not do the same as `erfa.s2c` if astrometric is a multidimensional with only some entries being near the origin.

It does suggest the representations should have a way to get unit vectors pointing in a coordinates position... Short of that, maybe
```
pco = astrometric.get_xyz(axis=-1)
at_origin = np.all(pco == 0, axis=-1, keepdims=True)
pco[at_origin] = [1., 0., 0.] << pco.unit
pco /= np.sqrt((pco**2).sum(-1, keepdims=True)
```

(need to think a bit more...)

Reply:
I'm thinking it may be easier not to pass in a CartesianRepresentation at all, but a SphericalRepresentation and retain the call to erfa.s2c....